### PR TITLE
Platformatica reference value edit test

### PR DESCRIPTION
**Pull request by Andrey2694**
TС_006.01 | Entity > Reference values > Input (New record)
TС_006.02 | Entity > Reference values > Edit (Existing record)

### CODE REVIEW

File: src/test/java/ReferenceValuesTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        getDriver().findElement(By.id("pa-entity-form-save-btn")).click();
+
+        Assert.assertEquals(LABEL_VALUE, getDriver().findElement(By.xpath("//tr[@data-index = '0']/td[2]/a")).getText());
+        Assert.assertEquals(FILTER1_VALUE, getDriver().findElement(By.xpath("//tr[@data-index = '0']/td[3]/a")).getText());
+        Assert.assertEquals(FILTER2_VALUE, getDriver().findElement(By.xpath("//tr[@data-index = '0']/td[4]/a")).getText());
+    }
+
+    @Test
+    void testEditRecord() {
+        testCreateRecord();
```
Comment (MariaMoroz) [inline src/test/java/ReferenceValuesTest.java:30]: Why aren't your tests public?

File: src/test/java/ReferenceValuesTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import base.BaseTest;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class ReferenceValuesTest extends BaseTest {
+    private static final String LABEL_VALUE = "Label";
```
Comment (MariaMoroz) [inline src/test/java/ReferenceValuesTest.java:9]: As I remember, all classes start with word "Entity"

File: src/test/java/ReferenceValuesTest.java (hunk 1 above)
Comment (Andrey2694) [reply src/test/java/ReferenceValuesTest.java:30]: for what? it is working without public 
and i use this method just in one class

File: src/test/java/ReferenceValuesTest.java (hunk 2 above)
Comment (Andrey2694) [reply src/test/java/ReferenceValuesTest.java:9]: ok ,will fix

File: src/test/java/ReferenceValuesTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        Assert.assertEquals(FILTER1_VALUE, getDriver().findElement(By.xpath("//tr[@data-index = '0']/td[3]/a")).getText());
+        Assert.assertEquals(FILTER2_VALUE, getDriver().findElement(By.xpath("//tr[@data-index = '0']/td[4]/a")).getText());
+    }
+
+    @Test
+    void testEditRecord() {
+        testCreateRecord();
+
+        getDriver().findElement(By.xpath("//tr[@data-index = '0']/td[last()]//button")).click();
+        getWait().until(ExpectedConditions.elementToBeClickable(By.xpath("//ul[@role = 'menu']//a[contains(text(), 'edit')]"))).click();
```
Comment (MariaMoroz) [inline src/test/java/ReferenceValuesTest.java]: Your code is not clear, how can I guess what it's the element with xpath //tr[@data-index = '0']/td[last()]//button?

Comment (Andrey2694) [reply src/test/java/ReferenceValuesTest.java]: fixed

File: src/test/java/ReferenceValuesTest.java (hunk 1 above)
Comment (Andrey2694) [reply src/test/java/ReferenceValuesTest.java]: ok chenged

File: src/test/java/EntityReferenceValuesTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    private static final String FILTER1_VALUE = "FILTER1";
+    private static final String FILTER2_VALUE = "FILTER2";
+
+    @Test
+    public void testCreateRecord() {
+        getDriver().findElement(By.xpath("//div[@id = 'menu-list-parent']//p[contains(text(), 'Reference values' )]")).click();
+        getDriver().findElement(By.className("card-icon")).click();
+
+        getDriver().findElement(By.id("label")).sendKeys(LABEL_VALUE);
+        getDriver().findElement(By.id("filter_1")).sendKeys(FILTER1_VALUE);
```
Comment (MariaMoroz) [inline src/test/java/EntityReferenceValuesTest.java:19]: If you use webelement more than 1 time, good practice is to create variable. The same thing for elements which you use in asserts

File: src/test/java/EntityReferenceValuesTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        getDriver().findElement(By.className("card-icon")).click();
+
+        getDriver().findElement(By.id(ID_LABEL)).sendKeys(LABEL_VALUE);
+        getDriver().findElement(By.id(ID_FILTER1)).sendKeys(FILTER1_VALUE);
+        getDriver().findElement(By.id(ID_FILTER2)).sendKeys(FILTER2_VALUE);
+        getDriver().findElement(By.id(ID_BUTTON_SAVE)).click();
+
+        Assert.assertEquals(LABEL_VALUE, getDriver().findElement(By.xpath(ASSERT_LOCATOR_LABEL)).getText());
+        Assert.assertEquals(FILTER1_VALUE, getDriver().findElement(By.xpath(ASSERT_LOCATOR_FILTER1)).getText());
+        Assert.assertEquals(FILTER2_VALUE, getDriver().findElement(By.xpath(ASSERT_LOCATOR_FILTER2)).getText());
```
Comment (MariaMoroz) [inline src/test/java/EntityReferenceValuesTest.java:34]: Sorry, just noticed: your assert is not correct. Syntax for TestNG Assertions: Assert.Method(actual, expected)

**Review by MariaMoroz — APPROVED** [review]

**Review by olhakovtun — APPROVED** [review]

**Pull request merged**